src/VestedRewardsDistributionJob.sol

```diff
@@ -0,0 +1,229 @@
+     */
+    function set(address dist, uint256 interval) external auth {
+        if (interval == 0) revert InvalidInterval();
+
+        if (!distributions.contains(dist)) distributions.add(dist);
+        intervals[dist] = interval;
+        emit Set(dist, interval);
+    }
+
+    /**
+     * @notice Removes `dist` from the job.
+     * @param dist The distribution contract.
+     */
+    function rem(address dist) external auth {
+        if (!distributions.remove(dist)) revert NotFound(dist);
+
+        delete intervals[dist];
+        emit Rem(dist);
+    }
+
+    /**
+     * @notice Checks if the job has the specified distribution contract.
+     * @param dist The distribution contract.
+     * @return Whether `dist` is set in the job or not.
+     */
+    function has(address dist) public view returns (bool) {
+        return distributions.contains(dist);
+    }
+
+    /**
+     * @notice Checks if the distribution is due for the specified contract.
+     * @param dist The distribution contract.
+     * @return Whether the distribution is due or not.
+     */
+    function due(address dist) public view returns (bool) {
+        // Gets the last time distribute() was called
+        uint256 last = VestedRewardsDistributionLike(dist).lastDistributedAt();
+        // If `last == 0` (no distribution so far), we allow it to be distributed immediately,
+        // otherwise, we can only distribute if enough time has elapsed since the last one.
+        if (last != 0 && block.timestamp < last + intervals[dist]) return false;
+
+        uint256 vestId = VestedRewardsDistributionLike(dist).vestId();
+        DssVestWithGemLike vest = VestedRewardsDistributionLike(dist).dssVest();
+        // Distribution is only due if there are unpaid tokens.
+        return vest.unpaid(vestId) > 0;
+    }
+
+    // --- Keeper Network Interface ---
+
+    /**
+     * @notice Executes the job though the keeper network.
+     * @param network The keeper identifier.
+     * @param args The arguments for execution.
+     */
+    function work(bytes32 network, bytes calldata args) external {
+        if (!sequencer.isMaster(network)) revert NotMaster(network);
+        if (args.length == 0) revert NoArgs();
+
+        (address dist) = abi.decode(args, (address));
+        // Prevents keeper from calling random contracts with a `distribute` method.
+        if (!has(dist)) revert NotFound(dist);
+        // Ensures that enough time has passed.
+        if (!due(dist)) revert NotDue(dist);
+
+        uint256 amount = VestedRewardsDistributionLike(dist).distribute();
+        emit Work(network, dist, amount);
+    }
+
+    /**
+     * @notice Checks if there is work to be done in the job.
+     * @param network The keeper identifier.
+     * @return ok Whether it should execute or not.
+     * @return args The args for execution.
+     */
+    function workable(bytes32 network) external override returns (bool ok, bytes memory args) {
```

Comment:
Just something to maybe note to the keeper network operators: ETH_CALL does have a gas limit providers use to prevent DoS. The limits are very high though (500M gas on Alchemy, 10x the current mainnet block gas limit on Infura, couldn't find the limit on Tenderly).

This function could revert if there are enough workable jobs (as it will work through all of them). With current amount of jobs this is highly unlikely (~220k gas per job). If they rely on their own nodes they should review what limit they are enforcing (if any).

Reply:
Thanks for the info, I was actually unaware of that. I'll add a note to NatSpec regarding this limitation.

> (~220k gas per job)
I wonder how did you get to this figure. Runnig `forge test --gas-report` gives me:
![image](https://github.com/user-attachments/assets/aaea34ef-46ca-4321-8d59-79f43b35c387)

In any case, this seems like a highly unlikely scenario, it would require hundreds or maybe thousands of jobs to reach the existing limits.